### ISSUE TITLE
LogiX snapper support

### ISSUE THREAD

**Issue by LeDrascol** [Neos-Metaverse/NeosPublic#1488]
## Is your feature request related to a problem? Please describe.
Parenting a snappable object to a snapper slot does not transform the object to the snapper's parameters. This can still be performed by resetting all transforms after re-parenting to or from a snap slot, but could be better with proper support.

## Relevant issues
N/A

## Describe the solution you'd like

Possible solutions:
* Option on Set Parent: "Apply Snapper Transforms"
* Logix node: Snap To Slot
* Logix node: Unsnap From Slot

These are only sugguestions and not requests.

## Describe alternatives you've considered
- Manually set transforms after reparenting a slot.
- Use ProbablePrime's tutorial video

## Additional context
Updated to use new formatting.

**Comment by ProbablePrime** [comment]
I wanted to share this process for snapping things around: https://youtu.be/O255_Jlevr4

Your idea should still be considered, I just wanted to provide a solution you could try today that I personally find quite neat.

**Comment by Enverex** [comment]
To feed on from Prime's example - resetting a single object requires a lot of work (at least 4 components plus more bits) which for each item is far too much work - there should be a single component that can perform this action without having to DIY a solution together yourself each time.

**Comment by ProbablePrime** [comment]
4 Components OR 4 Logix Nodes(Although the nodes can work for any slot)

**Comment by ProbablePrime** [comment]
We now have set transform nodes which do this a lot more easily. To move back to the main issue(I apologize for the distraction with my earlier comments)... I actually came across a use for this today.

Hamish and I are working on some simulations of actions within the body and snapping things to various cell receptors is a common action of multiple types of cells. 

With these nodes as LeDrascol asks, we can attach colliders to these cells and proteins and using collision events trigger snappers. 

These snappers with an an animation time set to higher than usual will basically simulate the process of biology where stuff floats around randomly until it slowly finds somewhere it can snap into. 

You can see something here ![](https://stke.sciencemag.org/highwire/filestream/202017/field_highwire_adjunct_files/0/3232006tr2Animation1.gif)

Which is really complicated(i'm not entirely sure what's going on), but can be broken down into basically:
- Randomized movement
- Snaps.

**Now, I can do randomized moved easily** but snapping things together is currently **only possible if the object has been grabbed by a user.** 

Adding these nodes means **other things can trigger snaps** and this cell protein simulator is a really good example but could be too technical of an example. So let me know if its not clear.

**Comment by Zyzyl** [comment]
Prime's example above notes that "snapping things together is currently only possible if the object has been grabbed by a user". More specifically, snapping objects together is only possible if a grabbable object is _released_ from being grabbed by a user. 

This reminded me of a situation I had a few months ago where I made a pair of tongs for picking up food objects (collision detection -> reparenting to tongs) but ran into the issue that when items were released (de-parenting) they couldn't be received by a GrabbableReceiverSurface because there was no real grab release action involved. I assume that, under the hood, there is some event triggered when a grabbable is released which various components can respond to - snappers included. Therefore it seems to me that snapping events could be indirectly triggered via LogiX if there were a LogiX node which could simulate a grabbable release for a given hierarchy. This would seem to solve the current issue as I understand it (i.e. moving slots around in the hierarchy doesn't trigger snapping events in the same way releasing from grab does) while opening additional possibilities too.
